features/shopify-markets-js-api.md

```diff
@@ -0,0 +1,340 @@
+| totalVariantOptions | Optional | Number | Total variant option levels displayed in the recommendation. Here variant level means, the selection of one variant value loads the immediate next variant and so on. More on this in the below sections. When not provided, actual product variant count is used. This value can be below the total actual product variants but not above that value. Currently, we support only a maxium of 5 variant option levels |
+| variantSelector | Optional | Object/VariantOptionSelector | Needed only when recommendations display variant options. More on this in the next section |
+
+The following table lists all the allowed mandatory and optional parameters of `VariantOptionSelector` Object. This parameter is required only when the recommendation displays variant options.
+
+| Parameter Name | Required/Optional | Type | Description |
+| :-------------------: | :---------------------: | :----------: | :-------------- |
+| titleElement | Optional | String |  CSS classname selector/Id of the element hosting the variant option title. Either this or `optionElement` selector must be provided. Both are not mandatory. Title element won't be updated when this option is ignored |
+| optionElement | Optional | String |  CSS classname selector of all the elements hosting the variant option values. This selector should be in-depth such that it directly points to the element hosting the option value. Either this or `titleElement` selector must be provided. Both are not mandatory. Option elements won't be updated when this option is ignored |
+| selectedElement | Optional | String | Must be provided when `optionElement` selector is provided. CSS selector/id of the element selected in the variant option. In case of multiple variant options, It's required to use the same selector across all the variant options to avoid any issues.  | 
+| position | Optional | Number | Must be provided when `optionElement` selector is provided. Numeric value of the order number/position of the variant option as declared in the Shopify admin product page. It's not required to display the elements, in the same order, in the recommendations, but it's requred to supply the appropriate position number and it should match the order in the Shopify admin |
+| dependentVariantSelector | Optional | Object/VariantOptionSelector | Selectors configuration for sub-level variants that changes based on the value selected in this primary variant. A maximum of 5 sub-level variants can be supported |
+
+#### Error reporting
+In case of any issues with the API call or element structuring, the API will promptly report the errors in the browser console. Please contact Nosto support in case you come across these issues.
+
+#### Limitations
+1. Only allows upto 5 sub-level variants
+2. Only allows 1 primary variants
+3. Strict structuring necessary. In other words, it's necessary to follow the rules defined in the description of each API parameters.
```

Comment:
Is this point necessary? I think it's pretty clear it won't work if you make a typo.

Reply:
So you mean the `productSectionElement` has some mandatory fields? In that case, isn't it better to improve validation using `validate.js`?